packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +570,655 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+      ..addRect(verticalDivider);
+
+    for (int i = 0; i < pressedButtonRects.length; i += 1) {
+      bkFillPath.addRect(pressedButtonRects[i]);
+    }
+
+    canvas.drawPath(
+      bkFillPath,
+      _buttonBkPaint,
+    );
+
+    // Create the pressed buttons background path and paint it.
+    final Path pressedBkFillPath = new Path();
+    for (int i = 0; i < pressedButtonRects.length; i += 1) {
+      pressedBkFillPath.addRect(pressedButtonRects[i]);
+    }
+
+    canvas.drawPath(
+      pressedBkFillPath,
+      _pressedButtonBkPaint,
+    );
+
+    // Create the dividers path and paint it.
+    final Path dividersPath = new Path()
+      ..addRect(horizontalDivider)
+      ..addRect(verticalDivider);
+
+    canvas.drawPath(
+      dividersPath,
+      _dividerPaint,
+    );
+  }
+
+  void _drawButtonBackgroundsAndDividersStacked(Canvas canvas, Offset offset) {
+    final Offset dividerOffset = new Offset(0.0, dividerThickness);
+
+    final Path bkFillPath = new Path()
+      ..fillType = PathFillType.evenOdd
+      ..addRect(Rect.largest);
+
+    final Path pressedBkFillPath = new Path();
+
+    final Path dividersPath = new Path();
+
+    Offset accumulatingOffset = offset;
+
+    final List<RenderBox> children = getChildrenAsList();
+    for (int i = 0; i < children.length; i += 1) {
+      final bool isButtonPressed = pressedButtons.contains(i);
+      final bool isPrevButtonPressed = pressedButtons.contains(i - 1);
+      final bool dividerNeeded = i == 0 || !(isButtonPressed || isPrevButtonPressed);
+
+      final Rect dividerRect = new Rect.fromLTWH(
+        accumulatingOffset.dx,
+        accumulatingOffset.dy,
+        size.width,
+        dividerThickness,
+      );
+
+      final Rect buttonBkRect = new Rect.fromLTWH(
+        accumulatingOffset.dx,
+        accumulatingOffset.dy + dividerThickness,
+        size.width,
+        children[i].size.height,
+      );
+
+      // If this button is pressed, then we don't want a white background to be
+      // painted, so we erase this button from the background path.
+      if (isButtonPressed) {
+        bkFillPath.addRect(buttonBkRect);
+        pressedBkFillPath.addRect(buttonBkRect);
+      }
+
+      // If this divider is needed, then we erase the divider area from the
+      // background path, and on top of that we paint a translucent gray to
+      // darken the divider area.
+      if (dividerNeeded) {
+        bkFillPath.addRect(dividerRect);
+        dividersPath.addRect(dividerRect);
+      }
+
+      accumulatingOffset += dividerOffset + new Offset(0.0, children[i].size.height);
+    }
+
+    // Draw all of the button backgrounds.
+    canvas.drawPath(
+      bkFillPath,
+      _buttonBkPaint,
+    );
+
+    // Draw the pressed button backgrounds.
+    canvas.drawPath(
+      pressedBkFillPath,
+      _pressedButtonBkPaint,
+    );
+
+    // Draw all of the dividers.
+    canvas.drawPath(
+      dividersPath,
+      _dividerPaint,
+    );
+  }
```

Comment:
I think I'd probably collapse lines 1193 to 1210 into three:
```dart
 canvas.drawPath(buttonBackgroundFillPath, _buttonBackgroundPaint);
 canvas.drawPath(pressedButtonBackgroundFillPath, _pressedButtonBackgroundPaint);
 canvas.drawPath(dividersPath, _dividersPaint);
```

With self-explanatory consistent variable names, there's no need for the comments. And the code all fits on one line, so no need to wrap. This ends up being much more readable.

Reply:
I made the change, but for what it's worth, I disagree about the readability.  We're now forcing the reader to parse the 3rd token in the line to interpret intent. I wouldn't say that's an improvement.